src/Database/LSMTree/Internal/Run/Index/Compact.hs

```diff
@@ -106,25 +107,26 @@ import           Database.LSMTree.Internal.Vector
   minimum key and maximum key on each page. As such, the index stores the
   key-interval @[minKey p, maxKey p]@ for each page @p@. @search1@ searches the
   vector for the key-interval that contains the search key, if it exists, and
-  returns the corresponding vector index as a page number.
+  returns the corresponding vector index as a page number. A design choice of
+  ours is that the search will __always__ return a page number, even if the
+  index could answer that the key is definitely not in a page (see
+  'indexSearches').
```

Comment:
I see how that makes things easier by preserving the length of the vector, but at the cost of a few unnecessary disk reads. They are probably very few, but do we have a good way of estimating the percentage (at least order of magnitude)?

Reply:
It depends on the false-positive rate and the distribution of keys. Previously, the compact index would only return `Nothing` if the search key comes before all the keys in the index (the ranges covered by the index extend indefinitely towards the end of the key range), which also only happens for false positives. So it's a fraction of the false-positive rate

@dcoutts and I talked about this during a call, and we think it's sufficient for our use case. We have enough memory to assign to filters to make the probability of doing unnecessary reads very small, and seeing as we have uniformly distributed keys it is also relatively unlikely that a search key comes before all keys in the index